src/utils/notifications.ts

```diff
@@ -0,0 +1,64 @@
+import { LocalNotifications } from "@capacitor/local-notifications";
+
+import { getNotificationDates } from "../state/CalculationLogics";
+import { Cycle } from "../data/ClassCycle";
+
+export const requestPermission = () => {
+  LocalNotifications.requestPermissions()
+    .then(({ display }) => {
+      if (display !== "granted") {
+        console.log("Notification permission not received");
+        return;
+      }
+      console.log("Notification permission granted");
+    })
+    .catch((err) => {
+      console.error("Error requesting notification permission:", err);
+    });
+};
+
+export const removeAllNotifications = () => {
+  LocalNotifications.removeAllDeliveredNotifications()
+    .then(() => {
+      console.log("Old notifications removed");
+    })
+    .catch((error) => {
+      console.error("Error removing notifications:", error);
+    });
+};
+
+export const createNotifications = (cycles: Cycle[]) => {
+  const notificationsId = new Uint16Array(2);
+  crypto.getRandomValues(notificationsId);
```

Comment:
Unfortunately getRandomValues doesn't guarantee no - repeatable value here, so we cant use this function for notificationId generation

As documentation says we should use int32 value here in the range -2147483648 and 2147483647
So, you can store current last id number of the storage and generate a new one with current+1 and update value in storage accordingly. When limit is exceeded we should reset value back to -2147483648